sds_data_manager/lambda_code/IAlirtCode/ialirt_data_query_api.py

```diff
@@ -24,6 +25,22 @@ class BadTimeError(Exception):
     pass
 
 
+class DecimalEncoder(json.JSONEncoder):
+    """Convert Decimals to floats."""
+
+    def default(self, obj):
+        """Override JSON encoding for Decimal values."""
+        if isinstance(obj, Decimal):
+            # - If the Decimal is an integer, return int
+            # - Otherwise, float rounded to 3 decimal places
+            if obj % 1 == 0:
+                return int(obj)
+            return round(float(obj), 3)
+
```

Comment:
Do you also want to do this for float data types as well to round them, or is everything coming in as Decimal so you'll never have raw floats?

```suggestion
            return round(float(obj), 3)
    if isinstance(obj, float):
        return super().default(round(obj, 3))
```